Dockerfile.multi_arch

```diff
@@ -0,0 +1,30 @@
+#
+# Cilium incremental build. Should be fast given builder-deps is up-to-date!
+#
+# cilium-builder tag is the Git SHA of the compatible build image commit.
+# Keeping the old images available will allow older versions to be built
+# while allowing the new versions to make changes that are not backwards compatible.
+#
+# The base image here should be multi-arched beforehand when used on non-amd64 platforms, such as arm64
+# platform
+#
+FROM quay.io/cilium/cilium-envoy-builder:bb7c198879e48ac01ab77e0f06084a53a386d9ca as builder
```

Comment:
I've created the `-dev` repository so this base image can be changed for
`quay.io/cilium/cilium-envoy-builder-dev:bb7c198879e48ac01ab77e0f06084a53a386d9ca`. Once we have the bazel for arch64 and the github actions we will be able to create a docker image for `quay.io/cilium/cilium-envoy-builder-dev` that has support for arch64 and amd64. :tada: